src/write.rs

```diff
@@ -83,21 +87,31 @@ pub struct FileOptions {
 }
 
 impl FileOptions {
-    #[cfg(feature = "flate2")]
+    #[cfg(all(not(feature = "bzip2"), not(feature = "libflate")))]
     /// Construct a new FileOptions object
     pub fn default() -> FileOptions {
         FileOptions {
-            compression_method: CompressionMethod::Deflated,
+            compression_method: CompressionMethod::Stored,
             last_modified_time: time::now(),
             permissions: None,
         }
     }
 
-    #[cfg(not(feature = "flate2"))]
+    #[cfg(all(not(feature = "libflate"), feature = "bzip2"))]
     /// Construct a new FileOptions object
     pub fn default() -> FileOptions {
         FileOptions {
-            compression_method: CompressionMethod::Stored,
```

Comment:
@mvdnes Question about the code here:

The conditional branch for `#[cfg(feature = "bzip2")]` was not covered here. The `Stored` compression is used for bzip2 feature. Does it not matter? 

I added the edge case in my refactor. But this is currently blocking me from compiling both `deflate` and `bzip2`.

Reply:
Yes, that is right. Altough supported here, bzip2 is not that common for zip files. That is why it must be explicitly set in `FileOption`, even when it is the only method available.

Reply:
Ok. As it is now, there is a default method for each of the cases. In the case where we build with both targets, I add an extra helper method for Bzip2. 

Would you prefer I change it back so that default case for Bzip2 is `CompressionMethod::Storage`? And there is no extra helper method when we compile both `deflate` and `bzip2` features?

Reply:
Yes, I prefer `Stored` to be the default even if bzip2 is available.

Reply:
@mvdnes I've addressed the feedback and cleaned up the diff. I think this is ready for a little more more review!